applications/integrationtests/mechanics/PlateWithHole.cpp

```diff
@@ -101,27 +50,17 @@ bool CheckSolution(NuTo::Structure& s)
         {
             auto numericStressNuTo = ipStress.col(iIP);
             Eigen::Vector3d numericStress(numericStressNuTo[0], numericStressNuTo[1], numericStressNuTo[5]);
-            auto analyticStress  = GetAnalyticSolution(ipCoords.col(iIP));
-            double error         = (numericStress - analyticStress).norm();
-            double relativeError = error / analyticStress.norm();
-            if (relativeError > 0.05)
-            {
-                std::cout << "relative error " << relativeError << '\t';
-                std::cout << "N " << numericStress.transpose() << '\t';
-                std::cout << "A " << analyticStress.transpose() << std::endl;
-                error = true;
-            }
-            maxerror = std::max(maxerror, relativeError);
+            auto analyticStress  = NuTo::Test::PlateWithHoleAnalytical::AnalyticStress(ipCoords.col(iIP));
+            auto error = (analyticStress - numericStress).norm() / analyticStress.norm();
+            BOOST_CHECK_SMALL(error, tolerance);
         }
     }
-    std::cout << "maxerror " << maxerror << std::endl;
-    return error;
 }
 
-//! @brief Imports a mesh file and builds the hessian and the internal gradient.
-int main(int argc, char* argv[])
+BOOST_AUTO_TEST_CASE(PlateWithHole)
 {
-    boost::filesystem::path binaryPath = std::string(argv[0]);
+    auto binary = boost::unit_test::framework::master_test_suite().argv[0];
+    boost::filesystem::path binaryPath = std::string(binary);
     binaryPath.remove_filename();
```

Comment:
This tests writes its output in a directory based on the current binary location. This location is passed via `argv[0]` - path and filename and the filename is removed (`remove_filename()`). In `master` this test was a normal main file and `argv[0]` was passed as an argument. Now, it is extracted from the `master_test_suite()`